src/components/CheckoutButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { loadStripe } from "@stripe/stripe-js";
+import { Button } from "@chakra-ui/react";
+
+// stripePromise object which is resolved in the component
+// for redirecting to checkout
+const stripePromise = loadStripe(
+    process.env.NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY,
+);
+
+/**
+ * CheckoutButton is a button component that redirects to the
+ * stripe checkout page
+ */
+export const CheckoutButton: React.FC = () => {
+    const handleClick = async () => {
+        const { sessionId } = await fetch("/api/checkout/session", {
+            method: "POST",
+            headers: {
+                "content-type": "application/json",
```

Comment:
Thoughts on creating helper functions for GET, POST, etc so we don't have to set header and other stuff each time? Can be done in a separate PR.

Reply:
Yeah - that's a good idea! We can do it in a separate PR.

Reply:
Created a task in the backlog. Can you leave a TODO in places where we're currently making a request?

Reply:
done!